File: plot/src/main/java/net/fwitz/math/plot/binary/dual/DualNumberFunctionPanel.java
package net.fwitz.math.plot.binary.dual;

import net.fwitz.math.binary.dual.DualNumber;
import net.fwitz.math.plot.binary.BinaryNumberFunctionPanel;

public class DualNumberFunctionPanel extends BinaryNumberFunctionPanel<DualNumber, DualNumber> {
    DualNumberFunctionPanel(DualNumberFunctionPlot plot) {
        super(DualNumber.class, DualNumber.class, plot);
    }

    @Override
    protected DualNumberFunctionRenderer<DualNumber> createRenderer(int width, int height) {
        return new DualNumberFunctionRenderer<>(
                this, valueType,
                width, height,
                DualNumberFunctionValuesFilters.getInstance());
    }
}

File: plot/src/main/kotlin/net/fwitz/math/plot/binary/dual/DualNumberFunctionPanel.kt
package net.fwitz.math.plot.binary.dual

import net.fwitz.math.binary.dual.DualNumber
import net.fwitz.math.plot.binary.BinaryNumberFunctionPanel

class DualNumberFunctionPanel internal constructor(
    plot: DualNumberFunctionPlot
) : BinaryNumberFunctionPanel<DualNumber, DualNumber>(
    DualNumber::class.java,
    DualNumber::class.java,
    plot
) {
    override fun createRenderer(width: Int, height: Int) = DualNumberFunctionRenderer(
        this, valueType,
        width, height,
        DualNumberFunctionValuesFilters
    )
}